rustls/src/error.rs

```diff
@@ -377,6 +380,82 @@ impl From<CertificateError> for Error {
     }
 }
 
+#[non_exhaustive]
+#[derive(Debug, Clone)]
+/// The ways in which a certificate revocation list (CRL) can be invalid.
+pub enum CrlError {
+    /// The CRL had a bad, or unsupported signature from its issuer.
+    BadSignature,
+
+    /// The CRL contained an invalid CRL number, or an invalid revoked certificate
+    /// serial number.
+    InvalidCrl,
```

Comment:
It feels odd that the comment has more specifics than the variant name. In general I wonder if it's useful that we converge the very specific webpki errors to a smaller/more general set here. It certainly makes for less code in rustls and maybe avoids overwhelming users with overly specific errors, but on the other hand throwing away the detail from the webpki errors seems unfortunate, maybe?

Reply:
> In general I wonder if it's useful that we converge the very specific webpki errors to a smaller/more general set here.

That's what I was trying to do here: squashing the invalid CRL number and invalid serial number errors into a more general "InvalidCrl" error.

Would it be less confusing if I simplified the comment to not mention the specifics?

Reply:
That would be more consistent at least. Still not sure why we don't propagate the full level of detail that webpki gives us to the rustls consumers.

Reply:
> Still not sure why we don't propagate the full level of detail that webpki gives us to the rustls consumers.

I switched to propagating the more specific errors.